src/foundry/client/pixi/sources/light-source.d.ts

```diff
@@ -105,106 +118,55 @@ declare global {
   }
 
-  interface LightAnimationConfiguration {
-    /** The human-readable (localized) label for the animation */
-    label: string;
-
-    /** The animation function that runs every frame */
-    animation: (this: LightSource, dt: number, animation: LightAnimationConfiguration) => void;
-
-    /** A custom illumination shader used by this animation */
-    illuminationShader: AdaptiveIlluminationShader;
-
-    /** A custom coloration shader used by this animation */
-    colorationShader: AdaptiveColorationShader;
-
-    /** A custom background shader used by this animation */
-    backgroundShader: AdaptiveBackgroundShader;
-
-    /** The animation seed */
-    seed?: number;
-
-    /** The animation time */
-    time?: number;
-
-    /** @defaultValue `null` */
-    type?: keyof typeof CONFIG.Canvas.lightAnimations | null;
-  }
-
   /**
    * A specialized subclass of the PointSource abstraction which is used to control the rendering of light sources.
+   * @param object  - The light-emitting object that generates this light source
    */
   class LightSource extends PointSource {
-    /** @param object - The light-emitting object that generates this light source */
-    constructor(object: InstanceType<ConfiguredObjectClassForName<"AmbientLight" | "Token">>);
-
-    /**
-     * The light or darkness container for this source
-     * @defaultValue `this._createMesh(AdaptiveBackgroundShader)`
-     */
-    background: PIXI.Mesh;
-
-    /**
-     * The light or darkness container for this source
-     * @defaultValue `this._createMesh(AdaptiveIlluminationShader)`
-     */
-    illumination: PIXI.Mesh;
-
-    /**
-     * This visible color container for this source
-     * @defaultValue `this._createMesh(AdaptiveColorationShader)`
-     */
-    coloration: PIXI.Mesh;
-
-    static override sourceType: "light";
+    constructor(object: AmbientLight | Token);
 
-    /**
-     * Strength of the blur for light source edges
-     * @defaultValue `3`
-     */
-    static BLUR_STRENGTH: number;
+    /** {@inheritdoc} */
+    static sourceType: "light";
 
     /**
      * Keys in the LightSourceData structure which, when modified, change the appearance of the light
-     * @internal
-     * @defaultValue
-     * ```javascript
-     * [
-     *   "dim", "bright", "gradual", "alpha", "coloration", "color",
-     *   "contrast", "saturation", "shadows", "luminosity"
-     * ]
-     * ```
      */
-    protected static _appearanceKeys: string[];
+    static readonly _appearanceKeys: [
+      "dim",
+      "bright",
+      "attenuation",
+      "alpha",
+      "coloration",
+      "color",
+      "contrast",
+      "saturation",
+      "shadows",
+      "luminosity"
+    ];
 
     /**
-     * The object of data which configures how the source is rendered
-     * @defaultValue `{}`
+     * The computed polygon which expresses the area of effect of this light source
      */
-    data: Partial<LightSourceData>;
+    los: PointSourcePolygon | PIXI.Polygon;
```

Comment:
Currently has error:
> Property 'los' in type 'LightSource' is not assignable to the same property in base type 'PointSource'.
  Type 'PointSourcePolygon | Polygon' is not assignable to type 'PointSourcePolygon | undefined'.
    Type 'Polygon' is missing the following properties from type 'PointSourcePolygon': origin, config, rays, _getBounds, and 4 more.ts(2416)

When PointSource is updated for v10x, this may resolve.